packages/flutter/test/material/app_bar_test.dart

```diff
@@ -446,11 +441,80 @@ void main() {
     expect(appBarHeight(tester), lessThan(initialAppBarHeight));
     expect(appBarHeight(tester), initialTabBarHeight);
 
-    // Scroll the not-pinned appbar back into view
+    // Scroll the floating-pinned appbar back into view
     controller.jumpTo(0.0);
     await tester.pump();
     expect(appBarIsVisible(tester), true);
     expect(appBarHeight(tester), initialAppBarHeight);
     expect(tabBarHeight(tester), initialTabBarHeight);
   });
+
+  testWidgets('SliverAppBar expandedHeight, floating with snap:true', (WidgetTester tester) async {
+    await tester.pumpWidget(buildSliverAppBarApp(
+      floating: true,
+      pinned: false,
+      snap: true,
+      expandedHeight: 128.0,
+    ));
+    expect(appBarIsVisible(tester), true);
+    expect(appBarY(tester), 0.0);
+
+    // Scroll to the middle of the list. The (floating) appbar is no longer visible.
+    final ScrollPosition position = tester.state<ScrollableState>(find.byType(Scrollable)).position;
+    position.animateTo(256.0, curve: Curves.linear, duration: const Duration(milliseconds: 500));
+    await tester.pumpAndSettle();
+    expect(appBarIsVisible(tester), false);
+    expect(appBarY(tester), lessThan(-128.0));
+
+    // Trigger the floating appbar snap animation.
+    position.animateTo(250.0, curve: Curves.linear, duration: const Duration(milliseconds: 500));
+    position.updateUserScrollDirection(ScrollDirection.forward); // ignore: INVALID_USE_OF_PROTECTED_MEMBER, since this is using a protected method for testing purposes
```

Comment:
use a gesture to scroll the scrollable rather than animating it and manually setting the scroll direction, it'll be more resilient (and will save us a few $1000 in the technical debt benchmark :-) ).